test/e2e/instrumentation/metrics.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package instrumentation
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/onsi/gomega"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2emetrics "k8s.io/kubernetes/test/e2e/framework/metrics"
+	e2enode "k8s.io/kubernetes/test/e2e/framework/node"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	"k8s.io/kubernetes/test/e2e/instrumentation/common"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"github.com/onsi/ginkgo/v2"
+)
+
+var _ = common.SIGDescribe("Metrics", func() {
+	f := framework.NewDefaultFramework("metrics")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+	var c, ec clientset.Interface
+	var grabber *e2emetrics.Grabber
+	ginkgo.BeforeEach(func(ctx context.Context) {
+		var err error
+		c = f.ClientSet
+		ec = f.KubemarkExternalClusterClientSet
+		gomega.Eventually(ctx, func() error {
+			grabber, err = e2emetrics.NewMetricsGrabber(ctx, c, ec, f.ClientConfig(), true, true, true, true, true, true)
+			if err != nil {
+				framework.ExpectNoError(err, "failed to create metrics grabber")
+			}
+			return nil
+		}, 5*time.Minute, 10*time.Second).Should(gomega.BeNil())
+	})
+
+	/*
+	   Release: v1.28
+	   Testname: Kubelet resource metrics
+	   Description: Should attempt to grab all resource metrics from kubelet metrics/resource endpoint.
+	*/
+	framework.ConformanceIt("should grab all metrics from kubelet /metrics/resource endpoint", func(ctx context.Context) {
```

Comment:
If this is a new test, it has to start as a non-conformance test and soak for a couple weeks so that we know it is not flaky. Flaky tests cannot be promoted to conformance.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md

Tests are allowed to be promoted from non-conformance to conformance during code freeze, so it shouldn't delay the feature to let it soak.

Reply:
Thanks @johnbelamaric , amended to be a non conformance test.